### PR TITLE
iotedge 1161 - Document IoT endpoint usage

### PR DESCRIPTION
**Pull request by LimaEchoCharlie**

### CODE REVIEW

File: docs/things-endpoint.md [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+    amURL=http://am.localtest.me:8080/openam
+    thingId=thingymabot
+    thingPassword=5tr0ngG3n3r@ted
+    ```
+
+### Authenticate the Thing
+
+```bash
+authResponse=$(curl \
+    --silent \
```
Comment (JacoJooste) [inline docs/things-endpoint.md:422]:
```suggestion
    --silent \
    --header 'Accept-API-Version: resource=2.0, protocol=1.0' \
```

File: cmd/pem-to-jwks/main.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	signer, err := frcrypto.ParsePEM(block)
+	if err != nil {
+		log.Fatal(err)
+	}
+	jwks, err := json.Marshal(
+		jose.JSONWebKeySet{
+			Keys: []jose.JSONWebKey{
+				{
+					Key:   signer.Public(),
+					KeyID: "test",
```
Comment (LimaEchoCharlie) [inline cmd/pem-to-jwks/main.go:60]: -> command line input

File: cmd/things-jwt/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+	signer, err := frcrypto.ParsePEM(block)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	n := opts.Nonce
+	if n == -1 {
+		// if no nonce has been supplied, use unix time
+		n = time.Now().Unix()
```
Comment (LimaEchoCharlie) [inline cmd/things-jwt/main.go:94]: @JacoJooste - what is the upper limit for a nonce?

Comment (JacoJooste) [reply cmd/things-jwt/main.go:94]: In AM it's processed as a Long, so 2^63-1.

Comment (LimaEchoCharlie) [reply cmd/things-jwt/main.go:94]: Excellent thank you!
By my back of an envelope calculation, this should be okay for the next 29 billion years. So no need to worry.

File: cmd/things-jwt/main.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	frcrypto "github.com/ForgeRock/iot-edge/v7/internal/crypto"
+	"github.com/ForgeRock/iot-edge/v7/internal/jws"
+	"github.com/jessevdk/go-flags"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+func createThingJWT(key crypto.Signer, url string, version string, n int64, custom map[string]interface{}) (string, error) {
+	opts := &jose.SignerOptions{}
```
Comment (LimaEchoCharlie) [inline cmd/things-jwt/main.go:37]: @JacoJooste 
I am edging towards adding the `"typ":"JWT"` header. According to the [spec](https://datatracker.ietf.org/doc/html/rfc7519#section-5.1) it is optional and we obviously allow it but for best practice we should probably add it. What do you think?

Comment (JacoJooste) [reply cmd/things-jwt/main.go:37]: If it's optional and we don't process it on AM then I would leave it out. It just increases the size unnecessarily and it's another thing to document and maintain in my opinion.

Comment (LimaEchoCharlie) [reply cmd/things-jwt/main.go:37]: Good point about the size. I will remove it from the authentication JWT for consistency.

**Review by JacoJooste — APPROVED** [review]

**Pull request merged**